### ISSUE TITLE
`ssrPath` and `ssrSearch` are not inherited from parent `Router` if `hook` is defined.

### ISSUE THREAD

**Issue by funmaker** [molefrog/wouter#471]
Hi. I'm not sure if this is expected behavior but defining custom hook in a nested Router makes it no longer pass ssrPath/Search from parent to the hook. This breaks SSR in my project because Links try to access `window.location`.
Also the link to [Router](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-hrefsfn-) in README.md ToC is [broken](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-).

Also I'd like to say this is a really great library, it makes routing so much simpler and more flexible than alternatives. Greak work!

**Comment by molefrog** [comment]
> Also the link to [Router](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-hrefsfn-) in README.md ToC is [broken](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-).

Thank you, just fixed that. 

> Hi. I'm not sure if this is expected behavior but defining custom hook in a nested Router makes it no longer pass ssrPath/Search from parent to the hook. This breaks SSR in my project because Links try to access `window.location`. Also the link to [Router](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-hrefsfn-) in README.md ToC is [broken](https://github.com/molefrog/wouter?tab=readme-ov-file#router-hookhook-parserfn-basebasepath-).

Yes, this is expected. The reason is that routers with different hooks usually subscribe to different sources of location, so things like base paths or ssr don't really make sense. For example, I can imagine an app that uses default browser location, but also contains another app with hash location for example for sub navigation (like tabs). What is your use case?

**Comment by funmaker** [comment]
>What is your use case?

I have an App component that is the root of all the regular react content/code, including the Router, Routes, etc. The client side entry simply calls `ReactDOM.hydrateRoot(rootEl, <App />);`. On the server side I use `ReactDOMServer.renderToString` and to make wouter work I wrapped App with Router like so `<Router ssrPath={req.originalUrl}><App /></Router>`. Everything worked as expected, until I provided a custom hook for the Router within App component(to support navigation blocking). I imagined that `ssrPath` will sort of propagate to nested routers, just like it did without the hook. But if that's an expected behavior I will close this issue.

For anyone facing similar issue: I used `useRouter` hook to manually propagate ssr* props like so:

```ts
export function CustomRouter({ children }: CustomRouterProps) {
  const parentRouter = useRouter();
  
  const useCustomLocation: BaseLocationHook = (...args) => {
    // custom hook...
  };
  
  return (
    <Router hook={useCustomLocation} ssrPath={parentRouter.ssrPath} ssrSearch={parentRouter.ssrSearch}>
      {children}
    </Router>
  );
}
```
